### PR TITLE
Improve and fix problems for running DevFile API tests

### PR DESCRIPTION
**Pull request by musienko-maxim**
### What does this PR do?
Fix and improve problems which were described in the https://issues.redhat.com/browse/CRW-5023 issue
```
During this work, I found the following problems:
* When we generate files from https://api.github.com/repos/eclipse-che/che-devfile-registry/contents/devfiles  We can often face rate limits for unauthorized requests. To increase limits, we can add a GitHub token with an auth header 
* When we use the Inbuld devfile registry on OpenShift and Kubernetes platform - the objects to devfiles URLs are different and we need to manage it.
* There is no way to set k8s custom namespace/OpenShift projects for running APIDevfiles tests
```

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/CRW-5023

### How to test this PR?
* If we use generating files using https://api.github.com/repos/eclipse-che/che-devfile-registry/contents/devfiles/ URL just set up envs., and run /che/tests/e2e/configs/sh-scripts/runDevfileAcceptanceTests.sh
* If we use the registry from a cluster the example of environments for running:
```
MOCHA_DIRECTORY=api;
NODE_TLS_REJECT_UNAUTHORIZED=0;
TEST_NAMESPACE=admin-che;
TS_API_TEST_KUBERNETES_COMMAND_LINE_TOOL=kubectl;
TS_PLATFORM=kubernetes;
TS_SELENIUM_BASE_URL=https://192.168.49.2.nip.io/;
USERSTORY=DevfileAcceptanceTestAPI

```
### PR Checklist

[As the author of this Pull Request, I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix or new feature ](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix-or-new-feature)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by chepullreq2 — APPROVED** [review]

File: tests/e2e/constants/BASE_TEST_CONSTANTS.ts [hunk 1]
```diff
@@ -53,15 +54,22 @@ export const BASE_TEST_CONSTANTS: {
...
-		return BASE_TEST_CONSTANTS.TS_SELENIUM_BASE_URL.includes('devspaces')
-			? 'devspaces'
+		return BASE_TEST_CONSTANTS.TEST_NAMESPACE.length > 0
+			? BASE_TEST_CONSTANTS.TEST_NAMESPACE
+			: BASE_TEST_CONSTANTS.TS_SELENIUM_BASE_URL.includes('devspaces')
+			? 'admin-devspaces'
 			: BASE_TEST_CONSTANTS.TS_SELENIUM_BASE_URL.includes('che')
 			? 'che'
-			: 'default';
+			: 'admin-che';
```
Comment (dmytro-ndp) [inline tests/e2e/constants/BASE_TEST_CONSTANTS.ts:72]: Shouldn't it be `TEST_NAMESPACE` constant, not hardcoded `admin-che`?

File: tests/e2e/constants/BASE_TEST_CONSTANTS.ts [hunk 2]
```diff
@@ -53,15 +54,22 @@ export const BASE_TEST_CONSTANTS: {
...
+	TEST_NAMESPACE: process.env.TEST_NAMESPACE || '',
+
 	/**
 	 * application name (DevSpaces or Che)
 	 */
 	TESTING_APPLICATION_NAME: (): string => {
-		return BASE_TEST_CONSTANTS.TS_SELENIUM_BASE_URL.includes('devspaces')
-			? 'devspaces'
+		return BASE_TEST_CONSTANTS.TEST_NAMESPACE.length > 0
+			? BASE_TEST_CONSTANTS.TEST_NAMESPACE
```
Comment (dmytro-ndp) [inline tests/e2e/constants/BASE_TEST_CONSTANTS.ts]: @musienko-maxim : this multi line statement is really hard to read, which could lead to mistakes in future.
What do you think about using if-statements?

File: tests/e2e/constants/API_TEST_CONSTANTS.ts [hunk 3]
```diff
@@ -18,9 +18,14 @@ export enum KubernetesCommandLineTool {
...
 	GIT_HUB_CHE_DEVFILE_REGISTRY_URL: string;
+	TS_GIT_API_AUTH_TOKEN: string;
 } = {
 	INBUILT_APPLICATION_DEVFILE_REGISTRY_URL: (): string => `${BASE_TEST_CONSTANTS.TS_SELENIUM_BASE_URL}/devfile-registry/devfiles/`,
-	GIT_HUB_CHE_DEVFILE_REGISTRY_URL: 'https://api.github.com/repos/eclipse-che/che-devfile-registry/contents/devfiles/'
+	GIT_HUB_CHE_DEVFILE_REGISTRY_URL: 'https://api.github.com/repos/eclipse-che/che-devfile-registry/contents/devfiles/',
+	/**
+	 * gitHub has a rate limit for unauthorized requests to GitHub API. We can prevent this problems using authorization token
+	 */
+	TS_GIT_API_AUTH_TOKEN: process.env.TS_API_TEST_DEV_WORKSPACE_LIST || ''
```
Comment (dmytro-ndp) [inline tests/e2e/constants/API_TEST_CONSTANTS.ts]: Seems to be a typo.

Shouldn't it be:
```
TS_GIT_API_AUTH_TOKEN: process.env.TS_GIT_API_AUTH_TOKEN || ''
```
?

**Review by dmytro-ndp — APPROVED** [review]
looks good to merge

**Pull request merged**